src/native/libs/System.Native/pal_io.c

```diff
@@ -341,10 +347,13 @@ intptr_t SystemNative_Dup(intptr_t oldfd)
     int result;
 #if HAVE_F_DUPFD_CLOEXEC
     while ((result = fcntl(ToFileDescriptor(oldfd), F_DUPFD_CLOEXEC, 0)) < 0 && errno == EINTR);
-#else
+#elif HAVE_F_DUPFD
     while ((result = fcntl(ToFileDescriptor(oldfd), F_DUPFD, 0)) < 0 && errno == EINTR);
     // do CLOEXEC here too
     fcntl(result, F_SETFD, FD_CLOEXEC);
+#else
+    // https://github.com/bytecodealliance/wasmtime/blob/main/docs/WASI-rationale.md#why-no-dup
```

Comment:
nit: please use a permalink here that includes the commit sha, or summarize the reason in a short comment so that we still know why even if that link goes away.